lib/foreman_tasks/engine.rb

```diff
@@ -69,6 +69,8 @@ class Engine < ::Rails::Engine
 
       config.to_prepare do
         ::Api::V2::HostsController.send :include, ForemanTasks::Concerns::HostsControllerExtension
+        ::PuppetclassesController.send :include, ForemanTasks::Concerns::EnvironmentsExtension
```

Comment:
Right, I have misread it completely before (haven't noticed the `Foreman::Controller::Environments` is a concern (I would love to see the modules in Foreman to follow the directory structure…). With this fact, I agree this is better.